lib/event.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+import socket
+
+from utilities.toUTC import toUTC
+
+
+class Event(dict):
+
+    # We set default vaules so that we can later
+    # create an alert around these, and know when events
+    # have to use defaults
+    DEFAULT_STRING = 'None'
+
+    def add_required_fields(self):
+        if 'receivedtimestamp' not in self:
+            self['receivedtimestamp'] = toUTC(datetime.now()).isoformat()
+        if 'utctimestamp' not in self:
+            self['utctimestamp'] = toUTC(datetime.now()).isoformat()
+        if 'timestamp' not in self:
+            self['timestamp'] = toUTC(datetime.now()).isoformat()
+        if 'mozdefhostname' not in self:
+            self['mozdefhostname'] = socket.gethostname()
+        if 'tags' not in self:
+            self['tags'] = []
+        if 'category' not in self:
+            self['category'] = self.DEFAULT_STRING
+        if 'hostname' not in self:
+            self['hostname'] = self.DEFAULT_STRING
+        if 'processid' not in self:
+            self['processid'] = self.DEFAULT_STRING
```

Comment:
While logically, one would think processid is an int, we treat it in many areas of the codebase as a string, which is why I'm using the default string here.

Reply:
Yes, this is because it's easier to have ES default to use all strings, unanalyzed than to attempt to pick the correct types per field (lots of numeric options) and we never do queries by ranges of process ID anyway..

Reply:
Let's keep it as string